src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> element;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+
+    }
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = tail = element = new Node(null, value, null);
+        } else {
+            Node<T> newNode = new Node(tail, value, null);
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkForEqualsIndex(index, size);
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = element = newNode;
+        } else if (index == 0) {
+            head.prev = newNode;
+            newNode.next = head;
+            head = newNode;
+        } else if (index == size) {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        } else {
+            Node<T> previuos = getItByIndex(index - 1);
+            newNode.prev = previuos;
+            newNode.next = previuos.next;
+            previuos.next = newNode;
+        }
+        size++;
+    }
+
+    private Node<T> getItByIndex(int i) {
```

Comment:
Приватные методы должны располагаться под публичными в классе (так как они вспомогательные)
```suggestion
    private Node<T> getNodeByIndex(int i) {
```